lib/provider.ex

```diff
@@ -192,13 +214,13 @@ defmodule Provider do
                }
              ) do
           {:ok, values} ->
-            Enum.each(values, fn {k, v} -> Provider.Cache.set(__MODULE__, k, v) end)
+            Provider.Cache.set(__MODULE__, Enum.to_list(values))
```

Comment:
Nit: since `values` is a map, it would be slightly better to use `Map.to_list`, to make things a bit clearer.